Source/DafnyCore/AST/Statements/Statements.cs

```diff
@@ -293,16 +305,42 @@ public abstract class AssignmentRhs : INode {
     Attributes = attrs;
   }
   public abstract bool CanAffectPreviouslyKnownExpressions { get; }
+
   /// <summary>
-  /// Returns the non-null subexpressions of the AssignmentRhs.
+  /// Returns all (specification and non-specification) non-null expressions of the AssignmentRhs.
   /// </summary>
-  public virtual IEnumerable<Expression> SubExpressions {
+  public IEnumerable<Expression> SubExpressions {
+    get {
+      foreach (var e in SpecificationSubExpressions) {
+        yield return e;
+      }
+
+      foreach (var e in NonSpecificationSubExpressions) {
+        yield return e;
+      }
```

Comment:
…and this sort of code like this:

```suggestion
      foreach (var e in SpecificationSubExpressions.Concat(NonSpecificationSubExpressions)) {
        yield return e;
      }
```